### ISSUE TITLE
Overriding loader styles in webpack

### ISSUE THREAD

**Issue by helguita** [jaredpalmer/razzle#297]
Hello,

I'm trying to override webpack configuration so I can add some options on css-loader and postcss-loader. However I run into an issue that says: 

```
./src/components/mycomponent/MyComponent.css
Module build failed: Syntax Error

(5:1) Unknown word

  3 | // load the styles
  4 | var content = require("!!../../../node_modules/css-loader/index.js?{\"importLoaders\":1,\"modules\":true,\"localIdentName\":\"[name]__[local]___[hash:base64:5]\",\"camelCase\":true}!../../../node_modules/postcss-loader/lib/index.js!./MyComponent.css");

```
after reading some issues over css-loader. It looks like the issue is that I'm overriding the styles that you already defined on razzle conf (see https://stackoverflow.com/questions/37031123/why-am-i-not-being-able-to-compile-sass-with-webpack) and that doesn't go well. I guess that's why Vue created vue-loader.

Given my scenario, should it be best just to fork your project and start from there?

**Comment by jaredpalmer** [comment]
You don't need to fork razzle, but instead totally replace the css loader or add a new one. Here's an example `razzle.config.js` of a project that adds SCSS support.

https://gist.github.com/jaredpalmer/0a91a7bd354b875b913c74f4b16125f7

**Comment by d4rky-pl** [comment]
@jaredpalmer I just spent an hour unsuccessfully trying to add Sass to my example Razzle project until I found this issue and your gist. Might be a good idea to add it to README to save others some time :)

**Comment by jaredpalmer** [comment]
@d4rky-pl Sorry about that. Perhaps an example `with-scss` would be even better?

**Comment by d4rky-pl** [comment]
@jaredpalmer oh yeah, it'd be perfect :) I was actually planning to release this config (probably tomorrow) with some tweaks and configurability as npm package so it can be used in a simliar fashion to `my-razzle-modifictions` from README :)

**Comment by imjared** [comment]
Hey @jaredpalmer, I'm having a similar issue where I'd like to overwrite the default CSS loader. After adding my own loader in `razzle.config.js`, I just end up with dupe CSS loading behavior. A `console.log` of my config.module.rules looks like this (shortened for relevant parts):

```javascript
module.exports = {
  modify: (config, {target, dev}, webpack) => {
    const appConfig = Object.assign({}, config);
   
    appConfig.module.rules.push({
    	test: /\.css$/,
    	use: [`style-loader`, `css?modules&importLoaders=1`, `postcss-loader`]
    });
  
    return config;
  }
}

// results in

[{
  test: /\.css$/,
  exclude: ['/Users/jaredlaser/sites/playground/razzle-spa/build'],
  use: [
    [Object]
  ]
}, {
  test: /\.css$/,
  use: ['style-loader',
    'css?modules&importLoaders=1',
    'postcss-loader'
  ]
}]
```

It looks like you might be doing [something similar here](https://gist.github.com/jaredpalmer/0a91a7bd354b875b913c74f4b16125f7#file-razzle-config-js-L35-L50) by looping on config, seeing if a rule matches what you're looking for, then modifying but other than that approach, is there something I'm missing about how to do a full overwrite? I've used the [Gatsby approach](https://www.gatsbyjs.org/docs/add-custom-webpack-config/) before with success where loaders are named and easily removed thanks to [webpack-configurator](https://github.com/lewie9021/webpack-configurator) and I'm hoping there's a similarly simple solution I'm missing.

**Comment by jaredpalmer** [comment]
I would use `Array.prototype.find` instead so you completely overwrite the css setup.

**Comment by d4rky-pl** [comment]
@imjared I've been collecting helpers for `razzle.config.js` and putting them in module called [razzle-config-utils](https://github.com/d4rky-pl/razzle-config-utils) so I don't have to write the same `find`/`findIndex` hacks over multiple projects. You can check it out and see if it helps :)

What you're trying here would be possible with:

```
const { modifyRule } = require('razzle-config-utils')

module.exports = {
  modify: (config, {target, dev}, webpack) => {
    const appConfig = Object.assign({}, config);
    modifyRule(appConfig, { test: /\.css$/ }, (rule) => {
      rule.use = [
        'style-loader',
        { loader: 'css-loader', options: { modules: true, importLoaders: 1 } }, 
        'postcss-loader'
      ]
    })
    return appConfig;
  }
}
```

BTW, you're mixing (very?) old webpack 1.x syntax with the new one - [css should be css-loader](https://webpack.js.org/guides/migrating/#automatic-loader-module-name-extension-removed) and [you should no longer specify your configuration as `?query&string` but use `options` instead](https://webpack.js.org/guides/migrating/#loader-configuration-is-through-options).

**Comment by imjared** [comment]
@d4rky-pl I had settled on the following:

```javascript
appConfig.module.rules.map((rule) => {
  if(!!`css`.match(rule.test)) {
    return {
      test: /\.css$/,
      use: [`style-loader`, `css?modules&importLoaders=1`, `postcss-loader`]
    };
  }
  return rule;
});
```

But I'll have to checkout your toolset since it sounds like you have a bit more experience here.

WRT the syntax, good to know. We're still using Webpack v1 on my day-to-day project so I'm trying to learn as I go into the great unknown here. I definitely appreciate the thoughtful reply and tip.

**Comment by fahamidev** [comment]
Hi.
I have the same problem and I've read the comments, but I did not find a solution
```
./node_modules/react-persian-datepicker/lib/styles/basic.css
Module build failed: Syntax Error 

(2:1) Unknown word

  1 | 
> 2 | var content = require("!!../../../css-loader/index.js?{\"modules\":true,\"sourceMap\":true}!./basic.css");
    | ^
  3 | 
  4 | if(typeof content === 'string') content = [[module.id, content, '']];

 WARNING  Failed to compile with 1 warnings

```
and `razzle.config.js` : 
```
'use strict';

module.exports = {
    modify: (baseConfig, { target, dev }, webpack) => {
        const appConfig = Object.assign({}, baseConfig);

        appConfig.module.rules.push({
            test: /\.css$/,
            use: [
                'style-loader',
                {
                    loader: 'css-loader',
                    options: {
                        modules   : true,
                        sourceMap : true,
                    },
                }
            ]
        });


    return appConfig;
  }
};
```
I think I've done the settings correctly. I think I've done the settings correctly.
I want to use css files inside the mod_modules file.But it does not work properly.
The settings for the css file are taken from the [razzle.config.js](https://gist.github.com/jaredpalmer/0a91a7bd354b875b913c74f4b16125f7) template

@jaredpalmer @d4rky-pl What is the solution to this problem?

**Comment by d4rky-pl** [comment]
@EhsanFahami you're adding a loader second time and so it compiles twice. First remove the existing rules for css files.

**Comment by fahamidev** [comment]
@d4rky-pl Thank you for your guidance
But I do not know exactly what to delete ?

If you mean, I do not need css settings. And the settings are done by default?! Well, when I delete my settings, I encounter the following error that apparently this error occurred because it could not use css files inside the node_modules
```
/home/ehsan/Desktop/with-redux/node_modules/react-persian-datepicker/lib/styles/basic.css:1
(function (exports, require, module, __filename, __dirname) { .calendarContainer {
                                                              ^
SyntaxError: Unexpected token .
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:607:28)
    at Object.Module._extensions..js (module.js:654:10)
    at Module.load (module.js:556:32)
    at tryModuleLoad (module.js:499:12)
    at Function.Module._load (module.js:491:3)
    at Module.require (module.js:587:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/ehsan/Desktop/with-redux/node_modules/react-persian-datepicker/lib/components/Calendar.js:255:11)

```

So I did not understand exactly what you mean. Thanks a lot for help .

**Comment by d4rky-pl** [comment]
Please create a minimal github repository that reproduces your problem :)

**Comment by fahamidev** [comment]
@d4rky-pl I built a repository inside github. And after ` npm install `, and after ` npm start `, you will be faced with Error.
I have installed a package called `  react-persian-datepicker `. This package requires the use of ` css ` from node_modules. And for this reason I need to set the ` modules: True ` in ` css-loader `. [My repository address](https://github.com/EhsanFahami/with-redux)

I really appreciate your guidance
